components/camel-jsonpath/src/main/java/org/apache/camel/jsonpath/JsonPathEngine.java

```diff
@@ -92,6 +103,16 @@ public JsonPathEngine(String expression, boolean writeAsString, boolean suppress
         this.hasSimple = simpleInUse;
     }
 
+    private ObjectMapper findRegisteredMapper(CamelContext context) {
+        if (context != null) {
+            Set<ObjectMapper> mappers = context.getRegistry().findByType(ObjectMapper.class);
```

Comment:
There is findSingleByType which is easier to use
org.apache.camel.spi.BeanRepository#findSingleByType